jenkins/cloud-custodian/Jenkinsfile

```diff
@@ -0,0 +1,121 @@
+#!groovy
+
+// This Jenkins pipeline take care of the clean-up process
+// of stale resources that may be left behind in BKPR Jenkins tests
+// (both manual or automated).
+// It uses https://github.com/cloud-custodian/cloud-custodian as the 
+// rules engine to control all of the different cloud resources.
+//
+// This job will lock all of the BKPR platforms prior running the
+// Cloud Custodian policies.
+
+properties([
+  pipelineTriggers([cron('59 23 * * 6')]),
+  parameters([
+    stringParam(name: 'CUSTODIAN_VERSION', defaultValue: '0.8.46.0', description: "Cloud Custodian tool version"),
+    stringParam(name: 'REMOTE_BRANCH', defaultValue: 'master', description: "Remote branch to pull Cloud Custodian policies from"),
+  ])
+])
+
+def buildRelDefaults = [
+  'AKS_REL': '1.15,1.16',
+  'EKS_REL': '1.15,1.16',
+  'GKE_REL': '1.15,1.16',
+  'GEN_REL': '1.16',
+]
+
+Void noLog(String cmd) {
+    sh('#!/bin/sh -e\n' + cmd)
+}
+
+Void runCloudCustodian (String envs, String cloud, String region) {
+  noLog("""
+    docker run \
+      -u \$(id -u):\$(id -g) \
+      -e "HOME=/home" \
+      ${envs} \
+      -v $WORKSPACE:/code \
+      -v /etc/passwd:/etc/passwd \
+      gcr.io/bitnami-images/sre-cloud-custodian:${env.CUSTODIAN_VERSION} \
+      run --region ${region} -f /code/cache -s /code/out /code/policies/${cloud}.yaml
+  """)
+}
+
+
+def lockBkprPlatforms(platforms, closure) {
+  if (platforms.size() > 1) {
+      lockBkprPlatforms(platforms[1..-1], { lock(platforms[0]) { closure() }})
+  } else {
+      lock(platforms[0]) { closure() }
+    }
+}
+
+
+// Use the JNLP-docker agent, which runs Docker-in-Docker
+node('jnlp-agent-docker') {
+
+  try {
+    stage("Pull the Cloud-Custodian container") {
+      withCredentials([file(credentialsId: 'gcr-bitnami-images-ro', variable: 'secretFilePath')]) {
+        noLog("docker login -u _json_key -p \"\$(cat '${secretFilePath}')\" https://gcr.io")
+        sh "docker pull gcr.io/bitnami-images/sre-cloud-custodian:${env.CUSTODIAN_VERSION}"
+      }
+    }
+    stage("Download the Custodian policy files") {
```

Comment:
Can't we make these files available "locally" in the same (Jenkins) workspace?

Reply:
changed, thxx